host/hackrf-tools/src/hackrf_biast.c

```diff
@@ -0,0 +1,222 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include <hackrf.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <string.h>
+
+#if defined(__GNUC__)
+	#include <unistd.h>
+	#include <sys/time.h>
+#endif
+
+#define USAGE              1
+#define INVALID_BIAST_MODE 2
+
+void usage()
+{
+	fprintf(stderr,
+		"\nhackrf_biast - enable/disable antenna power on the HackRF for compatibility\n");
+	fprintf(stderr,
+		"               with software that does not support this function\n\n");
+	fprintf(stderr, "Usage: \n");
+	fprintf(stderr, "  -h         Display this help\n");
+	fprintf(stderr,
+		"  -R         Reset all bias tee settings to device default.  When combined\n");
+	fprintf(stderr,
+		"             with -r/-t/-o, those settings will take precedence.\n\n");
+
+	fprintf(stderr,
+		"  [-b mode]  1=Enable bias tee immediately, 0=disable immediately\n");
+	fprintf(stderr,
+		"  [-r mode]  Set default bias tee power when device enters RX mode\n");
+	fprintf(stderr,
+		"  [-t mode]  Set default bias tee power when device enters TX mode\n");
+	fprintf(stderr,
+		"  [-o mode]  Set default bias tee power when device enters OFF mode\n");
+
+	fprintf(stderr,
+		"  [-d serial_number]  Specify serial number of HackRF device to configure\n\n\n");
+	fprintf(stderr, "The -r/-t/-o options support the following mode settings:\n\n");
+	fprintf(stderr, "  leave		do nothing when entering mode\n");
+	fprintf(stderr, "  on		disable bias tee when entering mode\n");
+	fprintf(stderr, "  off		enable bias tee when entering mode\n\n");
+	exit(USAGE);
+}
+
+void update_user_mode(
+	const char* direction_str,
+	const char* strarg,
+	hackrf_bool_user_settting* setting)
+{
+	int mode = 999; // Assume failure
+	if (strcmp((const char*) "off", strarg) == 0) {
+		mode = 0;
+	} else if (strcmp((const char*) "on", strarg) == 0) {
+		mode = 1;
+	} else if (strcmp((const char*) "leave", strarg) == 0) {
+		mode = 2;
+	}
+
+	if (mode < 0 || mode > 2) {
+		fprintf(stderr, "Invalid mode '%s' for %s\n", strarg, direction_str);
+		exit(INVALID_BIAST_MODE);
+	}
+
+	setting->do_update = true;
+	if (mode < 2) {
+		setting->change_on_mode_entry = true;
+		setting->enabled = mode;
+	} else {
+		setting->change_on_mode_entry = false;
+		setting->enabled = false;
+	}
```

Comment:
Now that the program is no longer taking numeric modes on the command line, I think this function would be much clearer if you just removed the `mode` variable entirely and just put all the effects in one if/else/else block, with a catchall `else` at the end for invalid values.
```suggestion
	if (strcmp("off", strarg) == 0) {
	        setting->do_update = true;
		setting->change_on_mode_entry = true;
		setting->enabled = false;
	} else if (strcmp("on", strarg) == 0) {
	        setting->do_update = true;
		setting->change_on_mode_entry = true;
		setting->enabled = true;
	} else if (strcmp("leave", strarg) == 0) {
	        setting->do_update = true;
		setting->change_on_mode_entry = false;
		setting->enabled = false;
	} else {
		fprintf(stderr, "Invalid mode '%s' for %s\n", strarg, direction_str);
		exit(INVALID_BIAST_MODE);
	}
```